### ISSUE TITLE
lstmcell GPU OOM

### ISSUE THREAD

**Issue by ghost** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#43]
thanks for your tutorial，it is so helpful for a beginner。But i encounter a problem in [this line](https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning/blob/master/eval.py#L107), 

code is `h, c = decoder.decode_step(torch.cat([embeddings, awe], dim=1), (h, c))`

when running the while loop, gpu memory increase gradually, and in tutorial code, batch size is 1，so there is no critical problem。But when i increase the batchsize to 32 or 64，and refer to your test code to write my own test code, GPU is Out of memory.

i can not understand why [this code](https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning/blob/master/eval.py#L107) can bring memory increasing. Why didn't I find this problem during the training? Can you explain this phenomenon?

thanks for your reply

**Comment by sgrvinod** [comment]
Hi @jj8401, this is because the inference with beam search I have implemented works only for one image at a time. Therefore, it always assumes there is only one image. If you have more than one image in the batch, you can see that [this line will be problematic](https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning/blob/903ebcc8b64d35592b65c00d56e109d68c3f709b/eval.py#L75).

I just added [a comment](https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning/blob/1553f3ace04e5fab4e8401936914dd092b025d70/eval.py#L52) to make this clear.

It doesn't happen during training because there is no actual inference with beam search during training, just Teacher Forcing.
